Sources/Noora/Noora.swift

```diff
@@ -58,6 +58,27 @@ public struct ErrorAlert: ExpressibleByStringLiteral {
 }
 
 public protocol Noorable {
+    func singleChoicePrompt<T: Equatable & CustomStringConvertible>(
+        question: TerminalText,
+        options: [T]
+    ) -> T
```

Comment:
Imho a better way to go about this is to provide a `Noorable` extension where you can specify default parameters. That way, the API is not restricted to this minimal version or you need to specify _all_ the parameters.

Something like:
```swift
extension Noorable {
    func singleChoicePrompt<T: Equatable & CustomStringConvertible>(
        title: TerminalText? = nil,
        question: TerminalText,
        options: [T],
        description: TerminalText? = nil,
        collapseOnSelection: Bool = true
    ) -> T {
      singleChoicePrompt(title: ...)   
    }
}
```

Reply:
I have noticed we have multiple definitions of `singleChoicePrompt` in the protocol already. I'd remove them all in favor of keeping a _single one_. All the variants can be derived from this extension approach.

Reply:
I'll refactor the code.